packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp

```diff
@@ -2258,11 +2258,13 @@ namespace MueLu {
       ArrayRCP<bool> AHBCrows;
       AHBCrows.resize(AH_->getRowMap()->getNodeNumElements());
       size_t dim = Nullspace_->getNumVectors();
+#ifdef HAVE_MUELU_KOKKOS_REFACTOR
       if (useKokkos_)
         for (size_t i = 0; i < BCdomainKokkos_.size(); i++)
           for (size_t k = 0; k < dim; k++)
             AHBCrows[i*dim+k] = BCdomainKokkos_(i);
       else
+#endif
         for (size_t i = 0; i < static_cast<size_t>(BCdomain_.size()); i++)
           for (size_t k = 0; k < dim; k++)
             AHBCrows[i*dim+k] = BCdomain_[i];
```

Comment:
When we have compiled with Kokkos, we have the runtime option to actually use Kokkos. When we haven't compiled with Kokkos, there is only one choice.

Reply:
Note the this pattern gets a bit uglier if for some reason you need the `{` and `}` because the `if(useKokkos_)` and associated `else` closes are not one liners...

Reply:
You mean like this? :-P
https://github.com/trilinos/Trilinos/blob/e58a9eeac87497e2d586bfbc58b50f54c0d7f2af/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp#L387-L430

Reply:
Yup, if you could add an `std::enable_if` in a conditional that would get you the Graal of hard to understand modern code (well only if the enable if is templated on a variadic template or a functor tag :p)

Reply:
I see. Looking at these few lines out of context, it looks like if Kokkos is turned on it will run both of those loops. Do we want to run both loops? Or I suppose `BCdomain_.size()` could be 0 if Kokkos is turned on?

Reply:
No, if kokkos is on at compile time, we have an if-else statement, and we only enter one of the branches.

Reply:
Oh I'm following now! I saw the preprocessor `#ifdef` statement but I didn't see the `if(useKokkos_)`. How embarassing. You added the preprocessor if because the build will fail if Kokkos refactor is off. :stuck_out_tongue: